kuksa-client/kuksa_client/grpc/aio.py

```diff
@@ -251,19 +255,23 @@ async def set(self, *, updates: Collection[EntryUpdate], **rpc_kwargs) -> None:
                 grpc.*MultiCallable kwargs e.g. timeout, metadata, credentials.
         """
         paths_with_required_type = self._get_paths_with_required_type(updates)
+        
         paths_without_type = [
             path for path, data_type in paths_with_required_type.items() if data_type is DataType.UNSPECIFIED
         ]
         paths_with_required_type.update(await self.get_value_types(paths_without_type, **rpc_kwargs))
         req = self._prepare_set_request(updates, paths_with_required_type)
+        metadata = []
+        metadata.append(("authorization", self.token))
+        rpc_kwargs.__setitem__("metadata", metadata)
```

Comment:
Instead of unconditionally adding/replacing the `metadata` argument, `"authorization"` should be added to it if exists, otherwise it should be created. Otherwise, other metadata will be overwritten if it exists.

Reply:
> But maybe then it would be an idea to create `authorization_header` in `authorize()`

That's the case already (except for the naming).

>  then just adding it to the rpc_kwargs if the authorization_header is not empty

That's also fine.

But the point here is that "metadata" shouldn't be overwritten if it exists. If it exists, then "authorization" should be added to it instead.

Reply:
You mean if something else is written in metadata it should not be overwritten only appended with the authorization header. Now I get it.